packages/esm-patient-forms-app/src/htmlformentry/html-form-entry-wrapper.component.tsx

```diff
@@ -18,17 +20,26 @@ const HtmlFormEntryWrapper: React.FC<HtmlFormEntryWrapperProps> = ({ closeWorksp
 
   // hide the headers and breadcrumbs of the O2 page on load
   const onLoad = useCallback(() => {
+    setIsIframeLoading(false);
+    iframeRef.current.contentWindow.addEventListener('beforeunload', () => setIsIframeLoading(true));
     const iframe = iframeRef.current;
     const dashboard = iframeRef.current.contentDocument;
     dashboard.querySelector('header')?.remove();
     dashboard.querySelector('.patient-header')?.remove();
     dashboard.querySelector('#breadcrumbs')?.remove();
     iframe.style.display = 'block';
     iframe.height = dashboard.body.scrollHeight + 50 + 'px'; // set iframe height to the height of the form, with some padding
+
+    const styleTag = dashboard.createElement('style');
+    styleTag.innerHTML = `@media screen {
+      body { background: white; }
+     }`;
+    dashboard.head.appendChild(styleTag);
   }, []);
 
   return (
     <div>
+      {isIframeLoading && <SkeletonText heading={true} />}
```

Comment:
I wasn't sure which progress indicator to use.  The loading spinner felt a bit to small and this appealed to me more, but if there is a best practice / convention that is important I can switch to whatever.

Reply:
I'd probably just try a really large version of the loading spinner, but I don't have a strong opinion on that.

Reply:
OK, I've changed it to the InlineLoading component which I know is used elsewhere.  Is this better?

[Screencast from 2025-05-02 12-32-55.webm](https://github.com/user-attachments/assets/6acad677-7c1d-4365-b859-ada9d3f1f1f8)